backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java

```diff
@@ -38,6 +39,27 @@ public Response setCommit(Commit commit, BlobFunction setBlobs, RepositoryFuncti
     }
   }
 
+  private String generateCommitSHA(String blobSHA, Commit commit) throws NoSuchAlgorithmException {
+
+    StringBuilder sb = new StringBuilder();
+    if (!commit.getParentShasList().isEmpty()) {
+      List<String> parentSHAs = commit.getParentShasList();
+      Collections.sort(parentSHAs); // TODO  EL/AJ to verify /optimize
+      sb.append("parent:");
+      parentSHAs.forEach(pSHA -> sb.append(pSHA)); // TODO  EL/AJ to verify /optimize
+    }
+    sb.append(":message:")
```

Comment:
Here if `commit.getParentShasList()` is empty then this linr should be 
```suggestion
    sb.append("message:")
```

Reply:
does not matter we anyway use it to create SHA, I added the keys for debugging purposes